### PR TITLE
feat: dropdown action

### PR DESCRIPTION
**Pull request by jsomsanith**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
There is no way to add a dropdown in the actions list

**What is the new behavior?**

ActionDropdown component.
Externalize custom tooltip into a component (TooltipTrigger)

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

* Change the Actions component to instanciate Action or ActionDropdown depending on prop
* Api has changed

### CODE REVIEW

**Comment by acateland** [comment]
two lint warn on Warning: Failed prop type: The prop `id` is required to make `Dropdown` accessible for users of assistive technologies such as screen readers.

**Comment by jsomsanith** [comment]
Fixed the warnings in unit tests

File: src/List/DisplayLarge/DisplayLarge.component.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, { PropTypes } from 'react';
 import classNames from 'classnames';
 
-import Actions from '../../Actions';
+import Actions from '../../Actions/Actions';
```
Comment (jmfrancois) [inline src/List/DisplayLarge/DisplayLarge.component.js]: I don't like this kind of imports. Can't we do something better ?

File: src/List/Toolbar/__snapshots__/Toolbar.snapshot.test.js.snap [hunk 2]
```diff
@@ -218,12 +218,9 @@ exports[`Toolbar should render add btn 1`] = `
       className="nav navbar-nav"
       role={null}>
       <button
-        activeHref={undefined}
-        activeKey={undefined}
-        className="navbar-btn btn btn-success"
+        className="btn btn-success"
```
Comment (jmfrancois) [inline src/List/Toolbar/__snapshots__/Toolbar.snapshot.test.js.snap]: without navbar-btn it will look & feel bad

File: src/index.js [hunk 3]
```diff
@@ -66,8 +66,9 @@ import {
 	Well,
 } from 'react-bootstrap';
 
-import Action from './Action';
-import Actions from './Actions';
+import Action from './Actions/Action';
+import ActionDropdown from './Actions/ActionDropdown';
+import Actions from './Actions/Actions';
 import AppHeaderBar from './AppHeaderBar';
```
Comment (jmfrancois) [inline src/index.js]: why not import { Action, ActionDropDown, Actions } from './Actions'; ?

File: stories/ActionDropdown.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+			<h2>Examples</h2>
+			<p>By default :</p>
+			<ActionDropdown {...myAction} />
+			<p>With hideLabel option</p>
+			<ActionDropdown
+				{...myAction}
+				hideLabel
+			/>
+		</div>
+	));
```
Comment (jmfrancois) [inline stories/ActionDropdown.js]: could you show the beahavior if items is empty ?

File: src/Actions/Action/Action.component.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import React, { PropTypes } from 'react';
+import {
+	Button,
+	OverlayTrigger,
+} from 'react-bootstrap';
+import TooltipTrigger from '../../TooltipTrigger';
+import CircularProgress from '../../CircularProgress';
+import Icon from '../../Icon';
+
+function getButtonProps(props) {
```
Comment (jsomsanith) [inline src/Actions/Action/Action.component.js]: As discussed quickly, I created an util/getPropsFrom that extract the Component's props + native ones.
For now the list of native ones is an const array with only className.

File: src/List/DisplayLarge/DisplayLarge.component.js (hunk 1 above)
Comment (jsomsanith) [inline src/List/DisplayLarge/DisplayLarge.component.js]: Now we import Action, Actions, ActionDropdown from '../Actions'

File: src/List/Toolbar/__snapshots__/Toolbar.snapshot.test.js.snap (hunk 2 above)
Comment (jsomsanith) [inline src/List/Toolbar/__snapshots__/Toolbar.snapshot.test.js.snap]: Fixed with your first comment's fix

File: src/index.js (hunk 3 above)
Comment (jsomsanith) [inline src/index.js]: done

File: src/Actions/ActionDropdown/__snapshots__/ActionDropdown.test.js.snap [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+        tabIndex="-1">
+        document 2
+      </a>
+    </li>
+  </ul>
+</div>
+`;
+
+exports[`ActionDropdown should render a button with icon and tooltip 1`] = `
+<div
```
Comment (jsomsanith) [inline src/Actions/ActionDropdown/__snapshots__/ActionDropdown.test.js.snap]: Don't know how to test the tooltip presence without testing .... the OverlayTrigger component from react-bootstrap. On mouseover, it append the tootip div to the body. But nothing except the mouseover on the button can indicate that the tooltip is here.
I'll just pass and only test my code : the render of the button with only an icon

File: stories/ActionDropdown.js (hunk 4 above)
Comment (jsomsanith) [inline stories/ActionDropdown.js]: Changed the behavior to add a disabled information (`No option`) item when list is empty

**Comment by charlesnguyen** [comment]
I get a text-decoration: underline; when mouse is over the action dropdown. Is it wanted?

bug

File: src/Actions/ActionDropdown/ActionDropdown.component.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+		<DropdownButton
+			title={title}
+			bsStyle={style}
+			role="button"
+			{...rest}
+		>
+			{
+				items.length ?
+				items.map((item, index) => (<MenuItem {...item} key={index}>{item.label}</MenuItem>)) :
+					<MenuItem disabled>No options</MenuItem>
```
Comment (charlesnguyen) [inline src/Actions/ActionDropdown/ActionDropdown.component.js]: what happend if the menu is to long and it goes off the screen?

Comment (jsomsanith) [inline src/Actions/ActionDropdown/ActionDropdown.component.js]: I don't know this has to be tested but this is out of this PR scope.
We simply use react-bootstrap dropdown

**Comment by jsomsanith** [comment]
@charlesnguyen the text-decoration on btn-link is set by the bootstrap theme. If needed, I think the change should be done in the theme.

**Pull request merged**